app/src/main/java/ch/epfl/sweng/eventmanager/repository/impl/FirebaseEventRepository.java

```diff
@@ -30,14 +33,27 @@
     public FirebaseEventRepository() {
     }
 
-    @Override public LiveData<List<Event>> getEvents() {
+    @Override
+    public LiveData<List<Event>> getEvents() {
         FirebaseDatabase fdB = FirebaseDatabase.getInstance();
         DatabaseReference dbRef = fdB.getReference("events");
 
-        return FirebaseHelper.getList(dbRef, Event.class);
+        return Transformations.switchMap(FirebaseHelper.getList(dbRef, Event.class), list -> {
+            MediatorLiveData<List<Event>> events = new MediatorLiveData<>();
+            List<Event> eventList = new ArrayList<>();
+            for (Event event: list){
+                events.addSource(getEventImage(event), img -> {
+                    event.setImage(img);
+                    eventList.add(event);
+                    events.setValue(eventList);
```

Comment:
Il faut `setValue` events à chaque itération ?

Reply:
Oui, car chaque event peut modifier la liste indépendamment et le set doit se faire dans le events.addSource(). https://developer.android.com/reference/android/arch/lifecycle/MediatorLiveData